### ISSUE TITLE
Cannot set daily usage quota in initial ARM template deployment

### ISSUE THREAD

**Issue by neilboyd** [Azure/azure-functions-host#7289]
#### Investigative information

- Timestamp: 2021-04-14T11:45:56.567808+00:00
- Function App version: 2.0
- Function App name: FuncApp-2ien2c3tw2k2u
- Function name(s) (as appropriate): n/a
- Invocation ID: 195d8600-bb84-48ad-ab8d-2d6c7ef57c3f
- Region: West Europe

#### Repro steps

1. Copy the [function app quickstart template](https://github.com/Azure/azure-quickstart-templates/blob/master/101-app-function/azuredeploy.json)
2. Add `dailyMemoryTimeQuota` property at [line 40](https://github.com/Azure/azure-quickstart-templates/blob/977f237b7e2018d959b50bb50edb90bea8212f9a/101-app-function/azuredeploy.json#L40)
3. Deploy with `az group deployment create -g $resourceGroup --template-file $armTemplate`

#### Expected behavior

I expect that after step 3, the app is created with `dailyMemoryTimeQuota` set in the app.

#### Actual behavior

After step 3, the app is created **without** `dailyMemoryTimeQuota` set in the app.

#### Known workarounds

Repeat step 3.
Or `az functionapp update -g $resourceGroup -n $functionAppName --set dailyMemoryTimeQuota=12460`.

#### Related information 

This is a duplicate of #3918, but that was closed due to lack of response from OP.

**Comment by v-bbalaiagar** [comment]
Hi @neilboyd, Thank you for reaching out to us, We will be checking the backend logs and update you with the findings shortly

**Comment by pragnagopa** [comment]
@mattchenderson - Can you help finding the right contact who can fix this issue?

**Comment by bjorncoltof** [comment]
Any news on this? We are in a scenario where we deploy functions apps with this setting quite often and we want to keep the number of ARM requests as low as possible as we have already run in to the situation of being throttled on ARM requests.

**Comment by balag0** [comment]
@bjorncoltof If your app using dynamic sku on Linux too, then this setting is not supported. 

Also this setting is not intended to handle the scenario you mentioned. It is purely to prevent getting billable events generated beyond certain amount per day, there is no guarantee on outbound calls or any other activities performed within the function app.

What type of throttles are you seeing from ARM?
Is it max requests / min or max requests / hour or max requests per client?

**Comment by neilboyd** [comment]
NB: the quickstart has [moved](https://github.com/Azure/azure-quickstart-templates/pull/9752) to [azuredeploy.json](https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.web/app-function/azuredeploy.json)

**Comment by bjorncoltof** [comment]
@balag0 In our scenario we deploy a great number of function apps. While deploying those we are being throttled by the ARM API, I don't know which of these limits we hit, but my point was mote that this bug makes us use more ARM requests then necessary to set dailyMemoryTimeQuota for the app, as after the arm template, we have to follow that up with more arm calls to get the setting in. I don't think this will the remove the hurdle completely regarding ARM api limits/throttling, but at least will help by removing the calls needed for the second step.

**Comment by balag0** [comment]
Got it, I think dailyMemoryTimeQuota should be set at the same level as clientAffinityEnabled setting and not inside the siteconfig section at Line 40. Can you give that a try?

**Comment by bjorncoltof** [comment]
@balag0 That is actually the location we have been using already, but as said it will not be set on the initial deployment.

**Comment by neilboyd** [comment]
The setting is not visible in the portal for Linux apps (I think it was previously, but it's not now).
This template is for a Windows function app: [azuredeploy.json](https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.web/function-app-create-dynamic/azuredeploy.json)

Add `dailyMemoryTimeQuota` as before:
![image](https://user-images.githubusercontent.com/1768373/136955716-15654488-8636-434a-9a7a-883e04043e87.png)

The setting is not set when deployed:
![image](https://user-images.githubusercontent.com/1768373/136966370-ca572434-ab71-43d6-8222-acd4940f920c.png)

Doing either of the workaround steps will set this setting.

**Comment by balag0** [comment]
Oh okay, The original issue mentioned Line 40, so wanted to make sure it is not set at that location. let me try this out and see what's going on. can you share a correlationid from any of the recent deployments if you have it?

**Comment by neilboyd** [comment]
@balag0 I just re-ran it. The correlation id is 7b079c1e-6bf0-48dc-9506-d36efcfa9b92

**Comment by neilboyd** [comment]
this doesn't seem to be happening any more
